File: app/models/business_formalisation.go
package models

import (
	"github.com/goravel/framework/database/orm"
)

type BusinessFormalisation struct {
	orm.Model
	orm.SoftDeletes
	SmeID                  int     `json:"sme_id" db:"sme_id"`
	HasBankAccount         bool    `json:"has_bank_account" db:"has_bank_account"`
	HasTaxClarification    bool    `json:"has_tax_clarification" db:"has_tax_clarification"`
	IsRegisteredForVat     bool    `json:"is_registered_for_vat" db:"is_registered_for_vat"`
	IsMemberOfAssociation  bool    `json:"is_member_of_association" db:"is_member_of_association"`
	IsAffiliated           bool    `json:"is_affiliated" db:"is_affiliated"`
	HasExportLicense       bool    `json:"has_export_license" db:"has_export_license"`
	HasAccessedBds         bool    `json:"has_accessed_bds" db:"has_accessed_bds"`
	AnnualTurnover         float64 `json:"annual_turnover" db:"annual_turnover"`
	EstimatedValueOfAssets float64 `json:"estimated_value_of_assets" db:"estimated_value_of_assets"`
	FormalisationScore     int     `json:"formalisation_score" db:"formalisation_score"`
	ComplianceScore        int     `json:"compliance_score" db:"compliance_score"`
	TeamStructureScore     int     `json:"team_structure_score" db:"team_structure_score"`
	FinancialScore         int     `json:"financial_score" db:"financial_score"`
	CreatedBy              *int    `json:"created_by" db:"created_by"`
	UpdatedBy              *int    `json:"updated_by" db:"updated_by"`
	DeletedBy              *int    `json:"deleted_by" db:"deleted_by"`
	IpAddress              *string `json:"ip_address" db:"ip_address"`
	UserAgent              *string `json:"user_agent" db:"user_agent"`
}

func (r *BusinessFormalisation) TableName() string {
	return "business_formalisation"
}

File: resources/js/types/business_formalisation.ts
// TypeScript interfaces for BusinessFormalisation entities and operations
import { BaseModel } from './crud';

// Core BusinessFormalisation interface matching the backend model
export interface BusinessFormalisation extends BaseModel {
  smeId: number;
  hasBankAccount: boolean;
  hasTaxClarification: boolean;
  isRegisteredForVat: boolean;
  isMemberOfAssociation: boolean;
  isAffiliated: boolean;
  hasExportLicense: boolean;
  hasAccessedBds: boolean;
  annualTurnover: number;
  estimatedValueOfAssets: number;
  formalisationScore: number;
  complianceScore: number;
  teamStructureScore: number;
  financialScore: number;
  createdBy?: number;
  updatedBy?: number;
  deletedBy?: number;
  deletedAt?: string;
  deleted_at?: string;
  ipAddress?: string;
  userAgent?: string;
}

// BusinessFormalisation creation data (matches BusinessFormalisationCreateRequest)
export interface BusinessFormalisationCreateData {
  smeId?: number; // Optional since it will be set from the SME creation context
  hasBankAccount: boolean;
  hasTaxClarification: boolean;
  isRegisteredForVat: boolean;
  isMemberOfAssociation: boolean;
  isAffiliated: boolean;
  hasExportLicense: boolean;
  hasAccessedBds: boolean;
  annualTurnover: number;
  estimatedValueOfAssets: number;
  // formalisationScore is excluded - it's calculated internally
}

// BusinessFormalisation update data (matches BusinessFormalisationUpdateRequest)
export interface BusinessFormalisationUpdateData {
  hasBankAccount?: boolean;
  hasTaxClarification?: boolean;
  isRegisteredForVat?: boolean;
  isMemberOfAssociation?: boolean;
  isAffiliated?: boolean;
  hasExportLicense?: boolean;
  hasAccessedBds?: boolean;
  annualTurnover?: number;
  estimatedValueOfAssets?: number;
  // formalisationScore is excluded - it's calculated internally
}

// Form errors interface for validation
export interface BusinessFormalisationFormErrors {
  hasBankAccount?: string;
  hasTaxClarification?: string;
  isRegisteredForVat?: string;
  isMemberOfAssociation?: string;
  isAffiliated?: string;
  hasExportLicense?: string;
  hasAccessedBds?: string;
  annualTurnover?: string;
  estimatedValueOfAssets?: string;
}

// Helper type for form state in the SME creation wizard
export interface FormalisationFormData {
  hasBankAccount: boolean;
  hasTaxClarification: boolean;
  isRegisteredForVat: boolean;
  isMemberOfAssociation: boolean;
  isAffiliated: boolean;
  hasExportLicense: boolean;
  hasAccessedBds: boolean;
  annualTurnover: string; // String for form input, will be parsed to number
  estimatedValueOfAssets: string; // String for form input, will be parsed to number
}

// Default values for form initialization
export const DEFAULT_FORMALISATION_DATA: FormalisationFormData = {
  hasBankAccount: false,
  hasTaxClarification: false,
  isRegisteredForVat: false,
  isMemberOfAssociation: false,
  isAffiliated: false,
  hasExportLicense: false,
  hasAccessedBds: false,
  annualTurnover: '',
  estimatedValueOfAssets: '',
};